src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from "react"
+import { Formik } from "formik"
+import { createCollection } from "Apps/CollectorProfile/Routes/Saves2/Mutations/createCollection"
+import createLogger from "Utils/logger"
+import {
+  Flex,
+  Button,
+  LabeledInput,
+  ModalDialog,
+  Spacer,
+  EditIcon,
+  Text,
+} from "@artsy/palette"
+import { useSystemContext } from "System"
+import {
+  CreateNewListValues,
+  CreateCollectionMutationResult,
+} from "Apps/CollectorProfile/Routes/Saves2/types"
+import { Media } from "Utils/Responsive"
+import { useTranslation } from "react-i18next"
+
+interface CreateNewListModalProps {
+  visible: boolean
+  onClose: () => void
+  onComplete: (result: CreateCollectionMutationResult) => void
+}
+
+const logger = createLogger(
+  "CollectorProfile/Routes/Saves2/Components/CreateNewListModal"
+)
+
+const CreateNewListModal: React.FC<CreateNewListModalProps> = ({
+  onClose,
+  onComplete,
+}) => {
+  const { t } = useTranslation()
+  const maxNameLength = 40
+  const { relayEnvironment } = useSystemContext()
+  const [formFieldError, setFormFieldError] = useState("")
+  const [remainingCharacters, setRemainingCharacters] = useState(maxNameLength)
+
+  const handleSubmit = async (values: CreateNewListValues) => {
+    if (!relayEnvironment) {
+      return null
+    }
+
+    const params: CreateNewListValues = {
+      name: values.name,
+    }
+
+    try {
+      const response = await createCollection(relayEnvironment, params)
+      const responseOrError = response.createCollection?.responseOrError
+
+      if (responseOrError?.collection) {
+        onComplete({ id: responseOrError?.collection?.internalID })
+      } else {
+        const error = (
+          responseOrError?.mutationError?.message ?? ""
+        ).toLowerCase()
+        setFormFieldError(
```

Comment:
Do we have a good pattern to follow for handling server-side errors? I am simply taking an error that Gravity returns and display it to a user. Wording is a little bit off, but can be changed on the Gravity side.

Currently it is `name already exists.`, but should be `You already have a list with this name.`

Reply:
If it's not a field specific error we just pop up a `Toast`